hack/v1migration/delete.sh

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
+
+WORKDIR=${1:?must specify working directory}
+echo "Using workdir: $WORKDIR"
+
+case "${2}" in
+"--dry-run" )
+  dry_run="--dry-run"
+  ;;
+"" )
+  # When doing a dry run, allow the script to continue after errors. Operations such as deleting a CRD are likely
+  # to fail since the CRs are not being deleted in a dry run.
+  set -e
+  ;;
+* )
+  echo "Only valid flag is \"--dry-run\""
+  exit 1
+esac
+
+if [[ -z $dry_run ]]
+then
+  source "$(dirname $0)/verify_scaledown.sh"
+  verify_all_scaled_down
+fi
+
+mkdir -p "$WORKDIR"
+
+if [[ -z $dry_run && \
+      "$(ls "$WORKDIR" 2>/dev/null)" ]]
+then
+  echo "WARNING: workdir ($WORKDIR) is not empty."
+  echo "Files in workdir may be overwritten."
+  read -r -p "Do you wish to proceed? [y/N] " response
+  if [[ ! "$response" =~ ^([yY]([eE][sS])?)$ ]]
+  then
+    exit 1
+  fi
+fi
+
+HIVE_TYPES=( checkpoints clusterdeployments clusterdeprovisionrequests clusterimagesets clusterprovisions clusterstates dnsendpoints dnszones hiveconfig selectorsyncidentityprovider selectorsyncset syncidentityprovider syncsetinstance syncset )
+readarray -t HIVE_TYPES <<< \
+  "$(oc get crd -o json | \
+    jq -r '.items[].spec | select((.group="hive.openshift.io") and (.version="v1alpha1")) | .names.plural')"
```

Comment:
Forgot to kill off this readarray so it's still happening. I've deleted locally to keep moving so not urgent.

Reply:
Doh!